lib/spack/spack/util/windows_registry.py

```diff
@@ -68,8 +69,19 @@ def _gather_subkey_info(self):
         sub_keys, _, _ = winreg.QueryInfoKey(self.hkey)
         for i in range(sub_keys):
             sub_name = winreg.EnumKey(self.hkey, i)
-            sub_handle = winreg.OpenKeyEx(self.hkey, sub_name, access=winreg.KEY_READ)
-            self._keys.append(RegistryKey(os.path.join(self.path, sub_name), sub_handle))
+            try:
+                sub_handle = winreg.OpenKeyEx(self.hkey, sub_name, access=winreg.KEY_READ)
+                self._keys.append(RegistryKey(os.path.join(self.path, sub_name), sub_handle))
+            except OSError as e:
+                if hasattr(e, "winerror"):
+                    if e.winerror == 5:
+                        # This is a permission error, we can't read this key
+                        # move on
+                        pass
+                    else:
+                        raise
+                else:
+                    raise
```

Comment:
Can you update the PR description with the qualitative supporting changes in `windows_registry` (like this error handling which presumably would have frequently caused issues before)?